shell/cloud-credential/harvester.vue

```diff
@@ -78,7 +78,40 @@ export default {
       this.$nuxt.$loading.finish();
 
       this.value.setData('kubeconfigContent', kubeconfigContent);
-    }
+    },
+
+    'value.decodedData.clusterId': {
+      handler() {
+        if (this.test() === true) {
```

Comment:
I think because these three handlers are identical they should probably refer to one function defined elsewhere eg a method 
```
emitValidation()
  if (this.test() === true) {
    this.$emit('validationChanged', true);
  } else {
    this.$emit('validationChanged', false);
  }
```

Then the handlers for `value.decodedData.clusterId` `isImportCluster` and `value.decodedData.kubeconfigContent` would look like:

```
 handler() {
   this.emitValidation();
 }
```

Reply:
Done. Thanks for your review. Good suggestion!